### PR TITLE
PR: Handle loading data from NpzFile instances and add an override flag to the load

### LINKED ISSUE #11596: Variable of npz file cannot be read if namespace has the same name
### What steps will reproduce the problem?

If variable name of npz exist in namespace, AttributeError is raised.

# not exist
normal

# exist
error

#### example file
[npz_data.zip](https://github.com/spyder-ide/spyder/files/4225403/npz_data.zip)

### Traceback
```python-traceback
  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder/plugins/variableexplorer/widgets/namespacebrowser.py", line 444, in import_data
    error_message = self.shellwidget.load_data(self.filename, ext)
  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder/plugins/ipythonconsole/widgets/namespacebrowser.py", line 125, in load_data
    timeout=CALL_KERNEL_TIMEOUT).load_data(filename, ext)
  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder_kernels/comms/commbase.py", line 542, in __call__
    call_dict, call_data, self._comm_id)
  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder/plugins/ipythonconsole/comms/kernelcomm.py", line 114, in _get_call_return_value
    call_dict, call_data, comm_id)
  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder_kernels/comms/commbase.py", line 420, in _get_call_return_value
    return self._sync_error(reply['value'])
  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder_kernels/comms/commbase.py", line 480, in _sync_error
    error_wrapper.raise_error()
  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder_kernels/comms/commbase.py", line 90, in raise_error
    raise self.etype(self)
Exception in comms call load_data:

  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder_kernels/comms/commbase.py", line 342, in _handle_remote_call
    buffer['call_kwargs'])

  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder_kernels/comms/commbase.py", line 353, in _remote_callback
    *call_args, **call_kwargs)

  File "/home/ok97465/anaconda3/lib/python3.7/site-packages/spyder_kernels/console/kernel.py", line 257, in load_data
    data[new_key] = data.pop(key)

AttributeError: 'NpzFile' object has no attribute 'pop'

```

## Versions

* Spyder version: 4.0.1 
* Python version: 3.7.4
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Linux 4.15.0-88-generic

### Dependencies

```
atomicwrites >=1.2.0         :  1.3.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
cloudpickle >=0.5.0          :  1.2.2 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
IPython >=4.0                :  7.11.1 (OK)
jedi =0.14.1                 :  0.14.1 (OK)
keyring                      :  None (OK)
nbconvert >=4.0              :  5.6.1 (OK)
numpydoc >=0.6.0             :  0.9.2 (OK)
pexpect >=4.4.0              :  4.8.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
psutil >=0.3                 :  5.6.7 (OK)
pygments >=2.0               :  2.5.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.7 (OK)
xdg >=0.26                   :  0.26 (OK)
zmq >=17                     :  18.1.0 (OK)
qdarkstyle >=2.7             :  2.8 (OK)
qtawesome >=0.5.7            :  0.6.1 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
rtree >=0.8.3                :  0.9.3 (OK)
sphinx >=0.6.6               :  2.3.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
watchdog                     :  None (OK)
cython >=0.21                :  0.29.14 (OK)
matplotlib >=2.0.0           :  3.1.2 (OK)
numpy >=1.7                  :  1.18.1 (OK)
pandas >=0.13.1              :  1.0.0 (OK)
scipy >=0.17.0               :  1.3.2 (OK)
sympy >=0.7.3                :  1.5.1 (OK)
```

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

When loading from a NpzFile it needs to be convert to a proper `dict`. 

Also, a message to ask for an override of the variables or to keep the old values (current behavior done by adding a suffix to any new variable i.e `var` -> `var000` ) is added.

A preview:

id

### Issue(s) Resolved

Fixes #11596 
Depends on spyder-ide/spyder-kernels#208

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/plugins/ipythonconsole/widgets/namespacebrowser.py [hunk 1]
```diff
@@ -119,10 +119,15 @@ def copy_value(self, orig_name, new_name):
 
     def load_data(self, filename, ext):
         try:
+            # TODO: Add dialog informing of possible override of variables
+            #       Check if user wants an override or wants to keep possible
+            #       all values
+            override = False
```
Comment (goanpeca) [inline spyder/plugins/ipythonconsole/widgets/namespacebrowser.py]: This code should go outside the exception. Code within exceptions should be in general as little as possible to be granular in  what we want to catch. Also in this case `override = False`is something that will never fail.

**Comment by goanpeca** [comment]
Let's add an underscore when the new variable is created

so `val1000`  -> `val1_000 `

**Comment by dalthviz** [comment]
A new preview with the underscore:

image

**Comment by goanpeca** [comment]
Awesome @dalthviz looks great. Let me take a look at the failing test to see what can be the issue, before we merge this one.

**Comment by goanpeca** [comment]
Hmmm the failing now seems to be another thing @dalthviz and @ccordoba12 ?

**Comment by ccordoba12** [comment]
@goanpeca, the last failure is a segfault but the previous ones are related to `test_conda_env_activation`, so please take a look at that asap. Thanks!

**Comment by dalthviz** [comment]
@goanpeca as far as I can tell (at least in this particular PR) seems like the `test_conda_env_activation` fails the three times it can be run with an error indicating that the process running the kernel can't be found (I think maybe here the tear-down of the test is failing, or somehow the console is trying to be restarted but failing?) and then the test suit tries to run again and a bunch of threading errors appear (which also I think are related with the kernel).

Something I just saw is that there is a message regarding the fail to restart the kernel:

```log
Restarting kernel... 
2020-02-26T04:10:41.6894893Z 
2020-02-26T04:10:41.6895315Z 
2020-02-26T04:10:41.6895606Z  
2020-02-26T04:10:41.6896692Z Kernel died, restarting
2020-02-26T04:10:41.6897214Z 
2020-02-26T04:10:41.6897673Z 
2020-02-26T04:10:41.6898000Z  
2020-02-26T04:10:41.6898517Z info_page
2020-02-26T04:10:41.6899053Z <!--
2020-02-26T04:10:41.6900016Z     kernel_error.html
2020-02-26T04:10:41.6900634Z     ~~~~~~~~~~~~~~~~~
2020-02-26T04:10:41.6901178Z 
2020-02-26T04:10:41.6901802Z     Kernel error template for the IPython Console
2020-02-26T04:10:41.6902412Z 
2020-02-26T04:10:41.6903132Z     :copyright: Copyright by the Spyder Project Contributors
2020-02-26T04:10:41.6903926Z     :license: MIT license
2020-02-26T04:10:41.6904508Z -->
2020-02-26T04:10:41.6905006Z 
2020-02-26T04:10:41.6905805Z <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
2020-02-26T04:10:41.6906736Z   "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
2020-02-26T04:10:41.6907418Z 
2020-02-26T04:10:41.6908550Z <html>
2020-02-26T04:10:41.6909090Z 
2020-02-26T04:10:41.6909849Z <head>
2020-02-26T04:10:41.6910810Z     <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
2020-02-26T04:10:41.6912774Z     <link rel="stylesheet" href="file:///D:\a\1\s\spyder\plugins\help\utils\static\css/default.css" type="text/css"/>
2020-02-26T04:10:41.6913819Z </head>
2020-02-26T04:10:41.6914337Z 
2020-02-26T04:10:41.6914839Z <body>
2020-02-26T04:10:41.6915650Z   <div class="panel panel-danger">
2020-02-26T04:10:41.6916526Z     <div class="panel-heading">
2020-02-26T04:10:41.6917365Z       <div class="panel-title">An error ocurred while starting the kernel</div>
2020-02-26T04:10:41.6918213Z     </div>
2020-02-26T04:10:41.6918825Z     <div class="panel-body">
2020-02-26T04:10:41.6925293Z       <tt>Traceback (most recent call last):<br>  File "C:\Miniconda\envs\spytest&#8209ï¿½\lib\runpy.py", line 193, in _run_module_as_main<br>    "__main__", mod_spec)<br>  File "C:\Miniconda\envs\spytest&#8209ï¿½\lib\runpy.py", line 85, in _run_code<br>    exec(code, run_globals)<br>  File "D:\a\1\s\external&#8209deps\spyder&#8209kernels\spyder_kernels\console\__main__.py", line 11, in <module><br>    start.main()<br>  File "D:\a\1\s\external&#8209deps\spyder&#8209kernels\spyder_kernels\console\start.py", line 287, in main<br>    import_spydercustomize()<br>  File "D:\a\1\s\external&#8209deps\spyder&#8209kernels\spyder_kernels\console\start.py", line 39, in import_spydercustomize<br>    import spydercustomize<br>  File "D:\a\1\s\external&#8209deps\spyder&#8209kernels\spyder_kernels\customize\spydercustomize.py", line 29, in <module><br>    from spyder_kernels.customize.spyderpdb import SpyderPdb<br>  File "D:\a\1\s\external&#8209deps\spyder&#8209kernels\spyder_kernels\customize\spyderpdb.py", line 23, in <module><br>    from IPython.core.inputtransformer2 import TransformerManager<br>ModuleNotFoundError: No module named 'IPython.core.inputtransformer2'<br></tt>
2020-02-26T04:10:41.6930783Z     </div>
2020-02-26T04:10:41.6931375Z   </div>
2020-02-26T04:10:41.6931758Z </body>
2020-02-26T04:10:41.6932209Z 
2020-02-26T04:10:41.6932858Z </html>

```

And the traceback of that error indicates an error with IPython:

```traceback
ModuleNotFoundError: No module named 'IPython.core.inputtransformer2'
```

Which was the one I had in spyder-ide/spyder-kernels#208 Maybe, somehow, the IPython version at the end of the installation of packages isn't `7.12.0` but `6.1.0` ?

**Comment by ccordoba12** [comment]
@dalthviz, please resync the spyder-kernels subrepo with the HEAD of its 1.x branch.

**Pull request merged**